crates/models/llama/src/lib.rs

```diff
@@ -64,39 +53,49 @@ impl KnownModel for Llama {
         // model-global weights
         let wte = tl.load("tok_embeddings.weight")?;
 
-        let backend = if params.should_offload(0) {
-            Backend::Gpu
-        } else {
-            Backend::Cpu
-        };
+        let backend = params.backend(0);
 
-        let norm = tl.offload("norm.weight", backend)?;
+        let norm = tl.load("norm.weight")?.transfer_to(backend)?;
 
-        let output = tl.offload("output.weight", backend)?;
+        let output = tl.load("output.weight")?.transfer_to(backend)?;
 
         let mut layers = Vec::new();
 
         for i in 0..hyperparameters.n_layer {
-            let backend = if params.should_offload(i) {
-                Backend::Gpu
-            } else {
-                Backend::Cpu
-            };
+            let backend = params.backend(i);
+
             let layer = Layer {
                 attention_norm: tl
-                    .offload(&format!("layers.{i}.attention_norm.weight"), backend)?,
-                wq: tl.offload(&format!("layers.{i}.attention.wq.weight"), backend)?,
-                wk: tl.offload(&format!("layers.{i}.attention.wk.weight"), backend)?,
-                wv: tl.offload(&format!("layers.{i}.attention.wv.weight"), backend)?,
-                wo: tl.offload(&format!("layers.{i}.attention.wo.weight"), backend)?,
-                ffn_norm: tl.offload(&format!("layers.{i}.ffn_norm.weight"), backend)?,
-                w1: tl.offload(&format!("layers.{i}.feed_forward.w1.weight"), backend)?,
-                w2: tl.offload(&format!("layers.{i}.feed_forward.w2.weight"), backend)?,
-                w3: tl.offload(&format!("layers.{i}.feed_forward.w3.weight"), backend)?,
+                    .load(&format!("layers.{i}.attention_norm.weight"))?
+                    .transfer_to(backend)?,
+                wq: tl
```

Comment:
hm okay, I was wrong, this is a bit unwieldy with the `transfer_to` calls. Can you try adding an `Option<Backend>` parameter to `load` and see how that looks?

Reply:
I don't know if adding a `Option<Backend>` to `load` would be better. As we still have to always define a value for it. With the `Tensor::transfer_to`  implementation the offloading is completely decoupled from the loader which i quite like. But i don't know how we could make this easier to use 🤔

Reply:
Yeah, good point. Maybe, instead of having individual calls, you instead have

`for tensor in &[...] { tensor.transfer_to(backend)?; }`

Another alternative would be defining `transfer_to` on `Layer` itself, so that you can transfer the entire layer as a unit without worrying about mixing loading and transfer logic.

Reply:
I don't know if that's better. If we do the `transfer_to` on a `Layer` we simply hide the `transfer_to` calls of each tensor behind another function. And we still have to implement it for each model as they use different `Layer` structs.

We also have to think about the multi GPU case. There we can offload certain tensors of a layer to all GPUs but some maybe need to be synced to a single GPU 🤔

Reply:
Alright, yeah. Leave it as-is for now then, we'll revisit once we have other models and/or multi-GPU support.